ui/tests/acceptance/secrets/backend/azure/azure-configuration-test.js

```diff
@@ -31,113 +33,427 @@ module('Acceptance | Azure | configuration', function (hooks) {
+          );
+        assert
+          .dom(GENERAL.infoRowValue('Subscription ID'))
+          .hasText('subscription-id', 'Subscription ID, a generic field, has been set.');
+        // cleanup
+        await runCmd(`delete sys/mounts/${path}`);
+      });
     });
 
-    test('it should show configuration with WIF options configured', async function (assert) {
-      const path = `azure-${this.uid}`;
-      const type = 'azure';
-      const wifAttrs = {
-        subscription_id: 'subscription-id',
-        tenant_id: 'tenant-id',
-        client_id: 'client-id',
-        identity_token_audience: 'audience',
-        identity_token_ttl: 720000,
-        environment: 'AZUREPUBLICCLOUD',
-      };
-      this.server.get(`${path}/config`, () => {
-        assert.ok(true, 'request made to config when navigating to the configuration page.');
-        return { data: { id: path, type, ...wifAttrs } };
+    module('edit', function (hooks) {
+      hooks.beforeEach(async function () {
+        const path = `azure-${this.uid}`;
+        const type = 'azure';
+        const genericAttrs = {
+          // attributes that can be used for either wif or azure account access type
+          subscription_id: 'subscription-id',
+          tenant_id: 'tenant-id',
+          client_id: 'client-id',
+          environment: 'AZUREPUBLICCLOUD',
+        };
+        this.server.get(`${path}/config`, () => {
+          return { data: { id: path, type, ...genericAttrs } };
+        });
+        await enablePage.enable(type, path);
+      });
+
+      test('it should not save client secret if it has NOT been changed', async function (assert) {
+        assert.expect(2);
+        await click(SES.configure);
+        const url = currentURL();
+        const path = url.split('/')[3]; // get path from url because we can't pass the path from beforeEach hook to individual test.
+        this.server.post(configUrl('azure', path), (schema, req) => {
+          const payload = JSON.parse(req.requestBody);
+          assert.strictEqual(
+            undefined,
+            payload.client_secret,
+            'client_secret is not sent if it has not been changed'
+          );
+          assert.strictEqual(
+            payload.subscription_id,
+            'subscription-id-updated',
+            'subscription_id is included with updated value in the payload'
+          );
+        });
+        await fillIn(GENERAL.inputByAttr('subscriptionId'), 'subscription-id-updated');
+        await click(GENERAL.enableField('clientSecret'));
+        await click(GENERAL.saveButton);
+        // cleanup
+        await runCmd(`delete sys/mounts/${path}`);
+      });
+
+      test('it should save client secret if it HAS been changed', async function (assert) {
+        assert.expect(2);
+        await click(SES.configure);
+        const url = currentURL();
+        const path = url.split('/')[3]; // get path from url because we can't pass the path from beforeEach hook to individual test.
```

Comment:
could we save the path as `this.path` in the beforeEach hook instead?

Reply:
Oish, I tried. It doesn't work for a couple reasons that I remember. If I set the path to something dynamic like `path-${this.uuid}` it changes before each test and then when i call it within the test it changes again. And setting it to something hard coded like`path-azure`, makes it a little harder to debug what path is being used in what tests and prone to flakiness even with cleanup after each run.